src/node.rs

```diff
@@ -12,6 +13,244 @@ pub enum Node {
     EmptySlot,
 }
 
+impl KeyValueStore for Node {
+    type Key = NibbleKey;
+    type Value = Vec<u8>;
+}
+
+impl Tree<Node> for Node {
```

Comment:
I imagine the naming around `Tree` and `Node` might be confusing for someone reading the code. I don't really have better suggestions, maybe `Node` and `MPTNode`, or `Tree` and `MPT`?

Reply:
I agree, at least for the `Node` part. I want to try binary trees and I will rename it `MPTree` at this stage. Don't want to make this PR longer than strictly necessary.